### PR TITLE
feat: fix serviceMonitor for additional daemonsets

### PR DESCRIPTION
**Pull request by sumo-drosiek**
### Checklist

Match correct label:

```
        app.kubernetes.io/name: RELEASE-NAME-sumologic-otelcol-logs-collector-linux
        app.kubernetes.io/app-name: RELEASE-NAME-sumologic-otelcol-logs-collector
```

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -649,7 +649,7 @@ sumologic:
           - port: metrics
         selector:
           matchLabels:
-            sumologic.com/app: otelcol-logs-collector
+            sumologic.com/app-name: otelcol-logs-collector
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:652]: Where does this label come from?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:652]: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/7c1008b7e8c68f3d27c1b7dd40cbe1b92adf8825/deploy/helm/sumologic/templates/logs/collector/otelcol/daemonset.yaml#L43

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:652]: But these selectors are for the Service, not the Daemonset.

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:652]: oh, I was trying figure out why e2e are failing :facepalm:

File: tests/helm/testdata/goldenfile/services_with_service_monitor_labels/services_with_service_monitor_labels/default.output.yaml [hunk 2]
```diff
@@ -1,136 +1,4 @@
 ---
-# Source: sumologic/templates/events/common/service.yaml
```
Comment (swiatekm-sumo) [inline tests/helm/testdata/goldenfile/services_with_service_monitor_labels/services_with_service_monitor_labels/default.output.yaml:2]: Why are these resources removed from this file? Were they unused, and regenerating the file caused them to disappear?

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:652]:
```suggestion
            sumologic.com/app: otelcol-logs-collector
```

File: tests/helm/testdata/goldenfile/services_with_service_monitor_labels/services_with_service_monitor_labels/default.output.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply tests/helm/testdata/goldenfile/services_with_service_monitor_labels/services_with_service_monitor_labels/default.output.yaml:2]: I think something is wrong and only one object is put into the generated file

**Review by perk-sumo — APPROVED** [review]
LGTM 🛠️

**Pull request merged**